include/ginkgo/core/matrix/dense.hpp

```diff
@@ -262,30 +267,99 @@ class Dense
 
     std::unique_ptr<LinOp> conj_transpose() const override;
 
+    /**
+     * Writes the transposed matrix into the given output matrix.
+     *
+     * @param output  The output matrix. It must have the dimensions
+     *                `gko::transpose(this->get_size())`
+     */
+    void transpose(Dense *output) const;
```

Comment:
Do these functions have to be `public` ? Maybe if we are using it in only internally, they can be `protected` ?